internal/registry/validate.go

```diff
@@ -127,3 +130,54 @@ func appendResult(results []apierrors.ManifestResult, r apierrors.ManifestResult
 
 	return results
 }
+
+// RewriteAnnotationsYaml unmarshalls the specified yaml file, appends the content and
+// converts it again to yaml.
+func RewriteAnnotationsYaml(filename string, content map[string]string) error {
+
+	metadata, err := getAnnotationFileContents(filename)
+	if err != nil {
+		return err
+	}
+
+	// Append the contents to annotationsYaml
+	for key, val := range content {
+		metadata.Annotations[key] = val
+	}
+
+	err = writeAnnotationFile(filename, metadata)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getAnnotationFileContents(filename string) (*registrybundle.AnnotationMetadata, error) {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	annotationsYaml := &registrybundle.AnnotationMetadata{}
+	if err := yaml.Unmarshal(f, annotationsYaml); err != nil {
+		return nil, fmt.Errorf("error parsing annotations file: %v", err)
+	}
+	return annotationsYaml, nil
+}
+
+func writeAnnotationFile(filename string, annotation *registrybundle.AnnotationMetadata) error {
+	// TODO: replace `gopkg.in/yaml.v2` with `sigs.k8s.io/yaml`. Operator registry
+	// defines annotations with yaml format (using gopkg-yaml) and k8s-yaml takes name
+	// of field in the struct as the value of key in yaml.
+	file, err := yaml.Marshal(annotation)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filename, []byte(file), 0644)
```

Comment:
Nit: use the existing mode if possible and default to rw since this file is typically generated with those perms.
```suggestion
	mode := os.FileMode(0666)
	if info, err := os.Stat(filename); err == nil {
		mode = info.Mode()
	}
	err = ioutil.WriteFile(filename, []byte(file), mode)
```